src/components/multi_select_state.js

```diff
@@ -167,6 +245,11 @@ const withMultiSelectState = WrappedComponent =>
           this.handleChange
         );
       }
+      this.setState({
```

Comment:
please remove. this should be done only in case if select all is clicked. 
1. you do this always 
2. you are using wrong items list for filtering.